src/checker.ts

```diff
@@ -315,7 +315,14 @@ export class LicenseChecker extends EventEmitter {
     localDirectory: string | null,
     ...parents: string[]
   ): Promise<void> {
-    const pj: PackageJson = ensurePackageJson(json);
+    const pj: PackageJson =
```

Comment:
Instead of the ternary, could we instead do something like:

```js
packageName = packageName || json.name;
const isWhitelisted = this.isPackageWhitelisted(packageName);
if (isWhitelisted) {
  json.version = semver.valid(json.version) ? json.version : '0.0.0'
  // any other corrections we may want to make in the future
}
```